enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,362 @@
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    # Prevent upgrades to OpenShift Version 4.9
+    operators.coreos.com/maxOpenShiftVersion: "4.8"
+```
+
+**OCP Labels in the image index**
+
+By using the label `LABEL com.redhat.openshift.versions`, it is possible to define range criteria of OpenShift catalogue(s) where the operator bundle should be published, see:
+
+```yaml
+FROM scratch
+
+LABEL operators.operatorframework.io.bundle.mediatype.v1=registry+v1
+LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
+LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
+LABEL operators.operatorframework.io.bundle.package.v1=your-operator
+LABEL operators.operatorframework.io.bundle.channels.v1=your-channel
+LABEL operators.operatorframework.io.bundle.channel.default.v1=your-channel
+LABEL com.redhat.openshift.versions: v4.5-v4.7
+
+COPY manifests /manifests/
+COPY metadata /metadata/
+```
+
+The versioning scheme available in the above OpenShift versions label accepts:
+
+- Min version (e.g `v4.5`)
+- Range (e.g `v4.5-v4.7`)
+- A specific version (e.g `=v4.6`)
+
+**Examples**
+
+| Label | Description |
+| ------ | ----- |
+| `LABEL com.redhat.openshift.versions:v4.5` | It means valid for OCP `4.5` and later versions |
+| `LABEL com.redhat.openshift.versions:=v4.5` | Valid **ONLY** for `4.5`
+| `LABEL com.redhat.openshift.versions:v4.5-v4.7` | Valid **ONLY** for `4.5`, `4.6` and `4.7` |
+
+However, note that nothing prevents a customer from taking a `4.6` catalog and installing it on a `4.8` ocp cluster (they may or may not actually be able to install the operators that exist in that catalog, of course).
+
+**CSV spec for minimal k8s version**
+
+The spec `minKubeVersion` describe what is the minimal version of Kubernetes that is supported by the operator bundle, see:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+...
+spec:
+  ... 
+  minKubeVersion: 1.11.0
+```
+
+The above features can be used alone and/or combined.
+
+#### Criteria valid which will not checked against to a specific OCP version
+
+The motivation for the following check is to ensure that operators authors understand that when the `maxOpenShiftVersion` and `minKubeVersion` is not defined it means allow to distribute the operator bundle for any OpenShift Catalog version available and then, tha the operator projects has no limitations regarding the supported OCP/Kubernetes versions which probably is not the common scenario.
+
+| Type | Description |
+| ----- | ----- |
+|warning|Check if we can found the `maxOpenShiftVersion` annotation or `minKubeVersion` spec in CSV. Then, if none of this options be filled raise a warning such as; `Your operator bundle is not specifying the OpenShift or Kubernetes version(s) which are supported by it. Ensure that it is installable in all possible provided options or configure it accordingly.`.
```

Comment:
@bparees, your concern raised in https://github.com/openshift/enhancements/pull/661#discussion_r579720717 is addressed here.